src/client.js

```diff
@@ -19,81 +19,148 @@ class TouchPortalClient extends EventEmitter {
-      newState.parentGroup = `${parentGroup}`;
-    }
-    this.send(newState);
-  }
-
-  createStateMany(states) {
-    const createStateArray = [];
-
-    if (states.length <= 0) {
-      this.logIt('ERROR', 'createStateMany : states contains no data');
-      throw new Error('createStateMany: states contains no data');
+  /**
+   * Internal helper method for states
+   * 
+   * @param {Object[Object]} states - Array of object containing id,
+   * desc, defaultValue, optionally parentGroup.
+   * @param {String} stateType - TouchPortal state type removeState
+   * or createState.
+   * 
+   * @return {Array<Object>}
+   */
+  stateHelper(states, stateType) {
+    const stateArray = [];
+    if (!(typeof states == "object" || states.length)) {
+      this.logIt('ERROR', `${stateType} has to be vaild object.`)
     }
-
     states.forEach((state) => {
-      if (this.customStates[state.id]) {
-        this.logIt('WARN', `createState: Custom state of ${state.id} already created`);
-      } else {
-        this.customStates[state.id] = state.desc;
-        const newState = {
-          type: 'createState',
-          id: `${state.id}`,
-          desc: `${state.desc}`,
-          defaultValue: `${state.defaultValue}`,
+      const stateObject = {
+        type: `${stateType}`,
+        id: `${state.id}`
+      };
+      if (`${stateType}` === "createState") {
+        if (this.customStates[state.id]) {
+          this.logIt('ERROR', `createState: Custom state of ${state.id} already created`);
+          throw new Error(`createState: Custom state of ${state.id} already created`);
+        };
+        stateObject.desc = state.desc;
+        stateObject.defaultValue = state.defaultValue
+        if (state.parentGroup) {
+          stateObject.parentGroup = `${state.parentGroup}`;
         };
-        if (state.parentGroup !== '' || state.parentGroup !== undefined) {
-          newState.parentGroup = `${state.parentGroup}`;
+        this.customStates[state.id] = state.desc;
+      } else if (`${stateType}` === "removeState") {
+        if (!this.customStates.hasOwnProperty(state.id)) {
+          this.logIt('ERROR', `removeState: Custom state of ${state.id} never created, so cannot remove it`);
+          throw new Error(`removeState: Custom state of ${state.id} never created, so cannot remove it`);
         }
-        createStateArray.push(newState);
+        delete this.customStates[state.id];
       }
-    });
+      stateArray.push(stateObject);
+    })
+    return stateArray;
+  }
+
+  /**
+   * createState() allows you to create a state during runtime.
+   * 
+   * @param {String} id - unquie state id
+   * @param {String} desc - description of the state
+   * @param {String} defaultValue - default value for the state
+   * @param {String} parentGroup - Optionally parentGroup
+   * 
+   * @return {void}
+   */
+  createState(id, desc, defaultValue, parentGroup) {
+    this.customStates[id] = desc;
```

Comment:
becaue in stateHelper we do this, we don't have to do this here.
```suggestion
   
```

Reply:
ah yeah. sorry